ckan/tests/factories.py

```diff
@@ -401,6 +401,32 @@ def _create(cls, target_class, *args, **kwargs):
         return obj
 
 
+class APIToken(factory.Factory):
+    """A factory class for creating CKAN API Tokens"""
+
+    class Meta:
+        model = ckan.model.ApiToken
+
+    name = "first token"
+
+    @classmethod
+    def _build(cls, target_class, *args, **kwargs):
+        raise NotImplementedError(".build() isn't supported in CKAN")
+
+    @classmethod
+    def _create(cls, target_class, *args, **kwargs):
+        if args:
+            assert False, "Positional args aren't supported, use keyword args."
+
+        context = {"user": _get_action_user_name(kwargs)}
+        data_dict = {"name": kwargs["name"], "user": kwargs["user"]["name"]}
+
+        token_create = helpers.call_action(
+            'api_token_create', context=context, **data_dict
+        )
```

Comment:
We still want to allow the `kwargs` to be passed to the action as data dict, but I see that we need to replace the `user` property, so can we try:

```suggestion

        target_user_name = _get_action_user_name(kwargs)
        context = {"user": target_user_name}
        kwargs["user"] = target_user_name

        token_create = helpers.call_action(
            'api_token_create', context=context, **kwargs
        )
```

The changes in the DataPusher tests are fine

Reply:
Thanks for your help @amercader